src/core/geometry/qgsgeometryutils.cpp

```diff
@@ -1366,14 +1366,19 @@ QPair<Qgis::WkbType, QString> QgsGeometryUtils::wktReadBlock( const QString &wkt
 {
   QString wktParsed = wkt;
   QString contents;
-  if ( wkt.contains( QLatin1String( "EMPTY" ), Qt::CaseInsensitive ) )
+  const QString empty { "EMPTY" };
+  if ( wkt.contains( empty, Qt::CaseInsensitive ) )
   {
-    const thread_local QRegularExpression sWktRegEx( QStringLiteral( "^\\s*(\\w+)\\s+(\\w+)\\s*$" ), QRegularExpression::DotMatchesEverythingOption );
-    const QRegularExpressionMatch match = sWktRegEx.match( wkt );
-    if ( match.hasMatch() )
+    const thread_local QRegularExpression whiteSpaces( "\\s" );
+    wktParsed.remove( whiteSpaces );
+    const int index = wktParsed.indexOf( empty, 0, Qt::CaseInsensitive );
+
+    if ( /*index != -1 && */index == wktParsed.length() - empty.length() )
```

Comment:
Why the commented code?

Reply:
I forgot to remove it. Since wkt, contains "EMPTY", `index != -1` will be always true.